contracts/upgradeable_contracts/BaseFeeManager.sol

```diff
@@ -0,0 +1,65 @@
+pragma solidity 0.4.24;
+
+import "../upgradeability/EternalStorage.sol";
+import "../libraries/SafeMath.sol";
+import "../IRewardableValidators.sol";
+
+
+contract BaseFeeManager is EternalStorage {
+    using SafeMath for uint256;
+
+    event FeeUpdated(uint256 fee);
+
+    function calculateFee(uint256 _value, bool _recover) external view returns(uint256) {
+        uint256 fee = getFee();
+        uint256 eth = 1 ether;
+        if (!_recover) {
+            return _value.mul(fee).div(eth);
+        }
+        return _value.mul(fee).div(eth.sub(fee));
+    }
+
+    function setFee(uint256 _fee) external {
+        uintStorage[keccak256(abi.encodePacked("fee"))] = _fee;
+        emit FeeUpdated(_fee);
+    }
+
+    function getFee() public view returns(uint256) {
```

Comment:
is it really public? my understanding is that we always will call it by `callcode`

Reply:
It is called by `callcode` on `RewardableBridge.getFee` and also called by `BaseFeeManager` on `calculateFee`.
https://github.com/poanetwork/poa-bridge-contracts/blob/d5b9bb5dcae9b793e31f5a1ad16d82a56cdcd63c/contracts/upgradeable_contracts/BaseFeeManager.sol#L13-L14

`public` modifier is the only one that would work for both cases

Reply:
ok. thanks!